Amazon.IonDotnet/Internals/Binary/ManagedBinaryWriter.cs

```diff
@@ -279,6 +270,29 @@ public override void AddTypeAnnotation(string annotation)
             this.userWriter.AddTypeAnnotationSymbol(token);
         }
 
+        /// <inheritdoc />
+        /// <summary>
+        /// This is supposed to close the writer and release all their resources.
+        /// </summary>
+        public override void Dispose()
+        {
+            this.Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        private void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                var lengthBuffer = this.userWriter?.GetLengthBuffer();
+                Debug.Assert(lengthBuffer == this.symbolsWriter.GetLengthBuffer(), "lengthBuffers do not match");
+                lengthBuffer?.Dispose();
```

Comment:
Shouldn't this happen in the `RawBinaryWriter`'s `Dispose` implementation?

Reply:
Could you elaborate on this?